tests/test_nbt_template.py

```diff
@@ -0,0 +1,259 @@
+import sys
+import os
+
+try:
+    import api
+except ModuleNotFoundError:
+    sys.path.insert(0, os.path.join(os.path.dirname(__file__)), "src")
+
+import unittest
+from api.nbt_template import (
+    TemplateLoader,
+    NBTStructure,
+    NBTListStructure,
+    NBTCompoundStructure,
+    NBTEntry,
+    NBTListEntry,
+    NBTCompoundEntry,
+)
+from test_utils import TESTS_DIR
+
+
+class NBTTemplateTestBaseCases:
```

Comment:
You probably don't need a class inside a class. A simpler solution is to make the base class not inherit from unittest.TestCase and make the classes that inherit from NBTTemplateTestCase also inherit from unittest.TestCase.
Unrelated, but in the long term we might want to consider to move to pytest rather than use unittest.

Reply:
Fair point, I'll make the change and I'll look into pytest